### PR TITLE
Introduce DAPI GRPC Core service

### PR DESCRIPTION
**Pull request by abvgedeika**

### CODE REVIEW

File: lib/services/dapi/txFilterStream/DapiTxFilterStreamOptions.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ class DapiTxFilterStreamOptions extends NodeJsServiceOptions {
   mergeWithDefaultOptions(...customOptions) {
     const defaultPorts = {
       port: this.getRandomPort(20002, 29998),
+      coreGrpsPort: this.getRandomPort(20002, 29998),
```
Comment (shumkov) [inline lib/services/dapi/txFilterStream/DapiTxFilterStreamOptions.js]: We are spinning up grpc core service in the core process.

**Comment by abvgedeika** [comment]
build failed because new dapi image should be issued

**Comment by shumkov** [comment]
@abvgedeika You need to set `coreGrpsPort` as `nativeGrpcPort` in DAPI Client.

**Comment by shumkov** [comment]
@abvgedeika Looks good. Let's write tests while we are waiting for the new docker image.

File: lib/services/dapi/core/DapiCore.js [hunk 2]
```diff
@@ -65,19 +84,33 @@ class DapiCore extends NodeJsService {
...
       try {
         const blockHeight = await this.dapiClient.getBestBlockHeight();
         if (blockHeight > -1) {
           starting = false;
         }
+        response = await checkHealth(request);
       } catch (error) {
         if (error.message === "DAPI RPC error: getBestBlockHeight: Error: No MNs in list. Can't connect to the network.") {
           starting = false;
+          response = await checkHealth(request);
```
Comment (shumkov) [inline lib/services/dapi/core/DapiCore.js]: That's looks bad... Don't you want to have two separate loops in two separate functions (`initializeJsonRpc`, `initializeGrpc`) and then call these functions in parallel from `initialize` method?

**Comment by shumkov** [comment]
@abvgedeika the new DAPI image is ready.

**Comment by shumkov** [comment]
@abvgedeika it looks good! We need to update tests as well.

**Comment by abvgedeika** [comment]
> @abvgedeika it looks good! We need to update tests as well.

what tests need to be updated?

**Comment by shumkov** [comment]
@abvgedeika hm.. we are not testing ports, just checking that service is running. Ok, in this case, that's it.

**Pull request merged**